spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -781,7 +800,11 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
             Object value = CatalystTypeConverters.convertToScala(catalystValue, field.dataType());
             values.put(field.name(), String.valueOf(value));
           });
-          return new SparkPartition(values, partition.path().toString(), format);
+
+          FileStatus fileStatus =
+              JavaConverters.seqAsJavaListConverter(partition.files()).asJava().get(0);
```

Comment:
Why does this use `partition.files()` instead of `partition.path()`?

Reply:
Because here partition is PartitionDirectory 
```
case class PartitionDirectory(values: InternalRow, files: Seq[FileStatus])
```
listFiles returns a Seq of PartitionDirectory
```
  def listFiles(
      partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Seq[PartitionDirectory]
```


Before my change, partition is PartitionPath
```
case class PartitionPath(values: InternalRow, path: Path)
```

Reply:
Great, thanks for the context! I assumed that it would use the same values.